### ISSUE TITLE
Proxy multiple domains

### ISSUE THREAD

**Issue by SamanthaAdrichem** [webpack/webpack-dev-server#1228]
> **Please note that this template is not optional.** If you proceed with this form,
please fill out _all_ fields, or your issue may be closed as "invalid."
Please do not delete this template. Please ask questions on StackOverflow or the
webpack Gitter (https://gitter.im/webpack/webpack). _General questions, how-to
questions, and support requests will be closed._
**Please do remove this header to acknowledge this message.**

* Operating System: Windws 7
* Node Version: 8.9.1
* NPM Version: 5.5.1
* webpack Version: 3.10
* webpack-dev-server Version: 2.9.5

- [x] This is a **bug**
- [x] This is a **feature** request
- [ ] This is a **modification** request
- [x] Request for help
### Code

```js
  // webpack.config.js
import { configPath } from "../vars/paths.config";

const webpackMerge = require('webpack-merge');
const webpack = require('webpack');
import { commonConfig } from './common.config';

const fs = require('fs');

export let developmentConfig;

developmentConfig = {
    devServer: {
        historyApiFallback: true,
        public : 'project.domain.ext.dev1:8080',
        disableHostCheck: true,
        https: {
            key: fs.readFileSync( configPath + '/resources/ssl/san_domain_com.key' ),
            cert: fs.readFileSync( configPath + '/resources/ssl/san_domain_com.crt-extensions' )
        },
        stats: {
            colors: true
        },
        proxy: {
            "https://login.domain.ext.dev1:8080/": {
                target: "https://login.domain.ext.dev01",
                secure: false,
                changeOrigin: true,
                bypass: bypassFunction,
                onProxyReq: relayRequestHeaders,
                onProxyRes: relayResponseHeaders
            }
        },
        hot: true,
        open: true,
        progress: true
    },
    plugins : [
        new webpack.HotModuleReplacementPlugin(),
        new webpack.NamedModulesPlugin()
    ]
};
developmentConfig = webpackMerge( commonConfig, developmentConfig );

function relayRequestHeaders(proxyReq, req) {}

function relayResponseHeaders(proxyRes, req, res) {
    if (proxyRes.headers && proxyRes.headers.location) {
        let expression = new RegExp('https://(project\\.(domain\\.ext|alt-domani\\.ext)\\.dev1)/', 'i');
        let match = proxyRes.headers.location.match( expression );
        if (match) {
            proxyRes.headers.location = proxyRes.headers.location.replace(match[1], match[1] + ':8080');
            res.append( 'location', proxyRes.headers.location );
        }
    }
}

function bypassFunction(req, res, proxyOptions) {
    if (req.headers && req.headers.host) {
        let expression = new RegExp('(project\\.(domain\\.ext|alt-domain\\.ext)\\.dev1)', 'i');
        let match = req.headers.host.match( expression );
        if (match && match.length > 0) {
            return '/index.html';
        }
    }
}
```

```js
  // additional code, remove if not needed.
```

### Expected Behavior
To only proxy if the domain matches login.domain.ext
To properly fetch style.css from login.domain.ext and not from project.domain.ext

### Actual Behavior
Proxies all URL's fixed it with weird bypass
Doesn't proxy CSS files, fetches wrong stylesheet

### For Bugs; How can we reproduce the behavior?
The bug is that the CSS file isn't passed through the bypass function
Nor does it seem possible to return a full domain path in the bypass function

### For Features; What is the motivation and/or use-case for the feature?
See: https://stackoverflow.com/questions/47769053/webpack-3-proxy-multiple-domains/47770261#47770261

Basically our code redirects to login for logging in, then login redirects back to the project. This is done with partial urls to prevent open redirects to other domains. We don't want fixes in the actual code to fix this, so proxy should do the works.

Maybe it already exists, but I can't find anything on the internet. It works okay with these hacks, except for the missing stylesheet., but I can't add a third domain. :(

**Comment by shellscape** [comment]
@SamanthaAdrichem thanks for checking in. A few things: an issue can't be a bug if a feature simply isn't supported, and we don't field support questions (requests for help) in the issues here.

That said, `webpack-dev-server` doesn't control how the proxying works, we leverage [`http-proxy-middleware`](https://github.com/chimurai/http-proxy-middleware). If you have a feature request, that's the proper place to contribute a new feature.
